rpcs3/Emu/RSX/Overlays/overlay_animation.cpp

```diff
@@ -23,49 +25,54 @@ namespace rsx
             }
         }
 
-        void animation_translate::update(u64 t)
+        void animation_translate::update(u64 frame)
         {
             if (!active)
             {
                 return;
             }
 
-            if (time == 0)
+            if (frame_start == 0)
```

Comment:
From the algorithm in the update function we can deduce some facts:
1. frame will never be == frame_end when updating animation deltas.
2. Animation step variable t will never be 1.0. This means the end frame will never correctly snap the element into the correct final position introducing a small subpixel error. There I propose the check for the end condition be made after the final 't' update, with clamping to 1.0. This way the final frame is guaranteed to execute correctly and snap the output to the correct frame position.

Reply:
That's already handled though isn't it? In `finish()` I do `current = end` which should be enough to ensure a proper position at the last and ensuing frames.

Reply:
Fair enough